### ISSUE TITLE
Autocomplete for types that implement Deref

### ISSUE THREAD

**Issue by chinedufn** [rust-analyzer/rust-analyzer#1183]
Hey!

Sorry if this is already tracked somewhere - I couldn't seem to find anything about it in the existing issues.

If you have something like..

```rust
impl Deref for FirstType {
  // ... derefs to SecondType
}

fn main () {
  let first_type: FirstType = FirstType::new();
  first_type.some_method_on_second_type(); // <-- does not auto complete
}
```

---

In short - autocomplete doesn't seem to currently suggest methods from `Deref`'d types.

I'd imagine that this is a known issue that's part of some other effort - but opening this issue just in case!

Cheers!

**Comment by matklad** [comment]
Hm, I think now, that https://github.com/rust-analyzer/rust-analyzer/pull/1154 is merged, we can actually implement this? 

cc @flodiebold , @marcogroppo

**Comment by flodiebold** [comment]
I don't think lang items are the main blocker for this; mainly it requires support for associated types. We could of course hack something together, but I've been waiting for Chalk integration to do it ;)

**Comment by marcogroppo** [comment]
@flodiebold this is a bit off topic, but I've noticed that we still don't complete `f32` and `f64`. They have non-macro based impls in libcore (actually there are two impls... `f32_runtime`...) so... they should work? It seems that we are not collecting them in `lang_items_query`... I'll try to investigate

**Comment by chinedufn** [comment]
@flodiebold (this is a beginner question)

I see this commit https://github.com/rust-analyzer/rust-analyzer/commit/b9c0c2abb79769852119dc9a595e63ee74eeba03 which I'm interpreting as being the same Chalk integration that you mentioned.

If that's the case - is this issue now possible but it's just lower priority compared to all of the other `rust-analyzer` stuff going on? Or were you saying that it is still blocked by future `Chalk` work?

Sorry for the rookie question - thanks so much!

**Comment by flodiebold** [comment]
@chinedufn It 'just' needs some work in rust-analyzer to actually implement this now. It's actually pretty high on my list ;)

**Comment by flodiebold** [comment]
This is actually done now.
